jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/MavenWebInfConfiguration.java

```diff
@@ -59,7 +60,28 @@ public void configure(WebAppContext context) throws Exception
                 LOG.debug("Setting up classpath ...");
             for (URI uri : jwac.getClassPathUris())
             {
-                loader.addClassPath(uri.toASCIIString());
+                // Not all Resource types supported by Jetty can be supported by WebAppClassLoader
+                String scheme = uri.getScheme();
+                if (scheme == null || scheme.equals("file"))
+                {
+                    // no scheme? or "file" scheme, assume it is just a path.
+                    loader.addClassPath(uri.getPath());
+                    continue;
+                }
+
+                if (scheme.equals("jar"))
+                {
+                    URI container = URIUtil.unwrapContainer(uri);
+                    if (container.getScheme().equals("file"))
+                    {
+                        // Just add a reference to the
+                        loader.addClassPath(container.getPath());
+                        continue;
+                    }
+                }
```

Comment:
Is this converting things like "jar:file:/usr/jetty/lib/somedependency.jar!/" back to just "/usr/jetty/lib/somedependency.jar"?

Are we to assume that we will never get anything like: "jar:file:/usr/jetty/lib/somedependency.jar!/WEB-INF/classes" or "jar:file:/usr/jetty/lib/somedependency.jar!/WEB-INF/lib/some.jar" ?

Reply:
@janbartel Can you give us a list of all the different uris that this method might see and what it is meant to do with each of them?   i.e. what is "the brief" of this method?